testing/test_column_line_numbers.py

```diff
@@ -0,0 +1,57 @@
+import ast
+from pathlib import Path
+from typing import List, Tuple
+
+import pytest
+from flake8_annotations import checker
+
+
+TEST_FILE = Path("./testing/code/column_line_numbers.py")
+
+# (line, column) tuples where we should get linting errors in the test file
+# Line numbers are 1-indexed
+# Column offsets are 0-indexed when yielded by our checker; flake8 adds 1 when emitted
+SHOULD_ERROR = ((16, 8), (16, 11), (24, 4), (25, 4), (26, 2))
+ERROR_CODE = Tuple[int, int, str, checker.TypeHintChecker]
+
+
+@pytest.fixture
+def parsed_errors(src_filepath: Path = TEST_FILE) -> List[ERROR_CODE]:
+    """Create a fixture for the error codes emitted by our testing code."""
+    with src_filepath.open("r") as f:
+        src = f.read()
+
+    tree = ast.parse(src)
+    lines = src.splitlines()
+
+    return list(checker.TypeHintChecker(tree, lines).run())
+
+
+def test_lineno(parsed_errors: List[ERROR_CODE]) -> None:
+    """
+    Check for correct line number values.
+
+    Emitted error codes are tuples of the form:
+      (line number, column number, error string, checker class)
+
+    Note: Line numbers are 1-indexed
+    """
+    should_error_lines = [coordinate_pair[0] for coordinate_pair in SHOULD_ERROR]
+    raised_error_lines = [error_tuple[0] for error_tuple in parsed_errors]
```

Comment:
It strikes me that this could be accomplished more efficiently by iterating both in parallel. `any()` could be used or each comparison could be asserted.

Reply:
Asserting each line/column would probably be easier to follow in the event that one fails than a potentially dense list of numbers